frontend/appflowy_flutter/lib/plugins/document/application/share_bloc.dart

```diff
@@ -1,65 +1,46 @@
-import 'dart:convert';
 import 'dart:io';
-import 'package:appflowy/plugins/document/application/share_service.dart';
-import 'package:appflowy/plugins/document/presentation/editor_plugins/parsers/divider_node_parser.dart';
-import 'package:appflowy/plugins/document/presentation/editor_plugins/parsers/math_equation_node_parser.dart';
-import 'package:appflowy/plugins/document/presentation/editor_plugins/parsers/code_block_node_parser.dart';
+import 'package:appflowy/workspace/application/export/document_exporter.dart';
 import 'package:appflowy_backend/protobuf/flowy-document2/entities.pb.dart';
 import 'package:appflowy_backend/protobuf/flowy-folder2/view.pb.dart';
 import 'package:appflowy_backend/protobuf/flowy-error/errors.pb.dart';
 import 'package:freezed_annotation/freezed_annotation.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
 import 'package:dartz/dartz.dart';
-import 'package:appflowy_editor/appflowy_editor.dart'
-    show Document, documentToMarkdown;
 part 'share_bloc.freezed.dart';
 
 class DocShareBloc extends Bloc<DocShareEvent, DocShareState> {
-  ShareService service;
-  ViewPB view;
-  DocShareBloc({required this.view, required this.service})
-      : super(const DocShareState.initial()) {
-    on<DocShareEvent>((event, emit) async {
-      await event.map(
-        shareMarkdown: (ShareMarkdown shareMarkdown) async {
-          await service.exportMarkdown(view).then((result) {
-            result.fold(
-              (value) => emit(
-                DocShareState.finish(
-                  left(_saveMarkdown(value, shareMarkdown.path)),
-                ),
-              ),
-              (error) => emit(DocShareState.finish(right(error))),
-            );
-          });
-
-          emit(const DocShareState.loading());
-        },
-        shareLink: (ShareLink value) {},
-        shareText: (ShareText value) {},
-      );
-    });
+  DocShareBloc({
+    required this.view,
+  }) : super(const DocShareState.initial()) {
+    on<ShareMarkdown>(_onShareMarkdown);
```

Comment:
Is there really a reason to not just put the one line of code here 😂 Instead of creating a private method.

Reply:
the other ones are unused now. It will look more clear to just put one event handler here.

Actually, I want to refactor all the event handlers into this pattern.

The other events are currently unused. IMO, it would be clearer to put the logic inside the single event handler. In fact, I intend to refactor all the event handlers into this pattern.

For example, 

Before
```dart
on<Event>(( ... {
  event.map(....) 
}))
```

After
```dart
on<EventA>(_onEventA);
on<EventB>(_onEventB);
```

Reply:
I agree with using the Events without freezed union, but whether the event handler is inside the `on<EventA>` or not doesn't make much sense, there's no reason to declare a private method, it's readable as is.